### PR TITLE
additional logging to get maximum token length of a sequence in the dataset

### PR DESCRIPTION
**Pull request by winglian**

### CODE REVIEW

File: src/axolotl/utils/trainer.py [hunk 1]
```diff
@@ -130,6 +124,16 @@ def process_datasets_for_packing(cfg, train_dataset, eval_dataset, tokenizer):
                         add_position_ids, num_proc=cfg.dataset_processes
                     )
 
+        if cfg.group_by_length or cfg.sample_packing:
```
Comment (NanoCode012) [inline src/axolotl/utils/trainer.py:127]: Is there a reason this output is only when these are set?

Comment (winglian) [reply src/axolotl/utils/trainer.py:127]: it's the only time they can be reliably calculated since it requires either the length column or position_ids column to calculate.

File: src/axolotl/utils/trainer.py [hunk 2]
```diff
@@ -130,6 +124,16 @@ def process_datasets_for_packing(cfg, train_dataset, eval_dataset, tokenizer):
...
+            max_input_len = np.max(get_dataset_lengths(train_dataset))
+            LOG.debug(f"max_input_len: {max_input_len}", main_process_only=True)
+
+        train_dataset = train_dataset.filter(drop_long, num_proc=cfg.dataset_processes)
```
Comment (NanoCode012) [inline src/axolotl/utils/trainer.py:131]: Is it possible to count how many samples are dropped?

Comment (winglian) [reply src/axolotl/utils/trainer.py:131]: yeah, let's slate that for another feature/PR. This was just shuffling around the order.

**Review by NanoCode012 — APPROVED** [review]
